src/components/RedefinedDomainResolver.tsx

```diff
@@ -0,0 +1,103 @@
+import React, {useCallback, useEffect, useState} from "react";
+import _debounce from 'lodash/debounce';
+import styled from "styled-components";
+import DropDown from "./DropDown";
+import {ContainerProps, RedefinedDomainResolverProps, InputProps} from "../types";
+import companyLogo from '../assets/small-logo.svg';
+import {baseStyle} from "../styles/baseStyle";
+import GlobalStyle from "../styles/globalStyle";
+import {RedefinedResolver} from "redefined-resolver";
+
+const RedefinedDomainResolver = (props: RedefinedDomainResolverProps) => {
+  const {width, height, onSelect} = props;
+  const [dropDownActive, setDropDownActive] = useState(false);
+  const [domain, setDomain] = useState("");
+  const [addresses, setAddresses] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
+
+  useEffect(() => {
+    domain.length === 0 && setDropDownActive(false);
+  }, [domain])
+
+  const handleDebounceFn = async (value: string) => {
+    if (value.length > 0) {
+      setDropDownActive(true);
+      setLoading(true);
+      setAddresses([]);
+      setError("");
+      await new RedefinedResolver().resolve(value)
```

Comment:
try catch?😅

Reply:
Fii
`then catch` is not actual and in this case you can write without them)

Reply:
ok done